packages/webview_flutter/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterDownloadListener.java

```diff
@@ -0,0 +1,24 @@
+package io.flutter.plugins.webviewflutter;
+
+import android.webkit.DownloadListener;
+import android.webkit.WebView;
+
+/** DownloadListener to notify the {@link FlutterWebViewClient} of download starts */
+public class FlutterDownloadListener implements DownloadListener {
+    final private FlutterWebViewClient webViewClient;
+    private WebView webView;
+
+    public FlutterDownloadListener(FlutterWebViewClient webViewClient){
+        this.webViewClient = webViewClient;
+    }
+
+    /** Sets the {@link WebView} that the result of the navigation delegate will be send to. */
+    public void setWebView(WebView webView){
+        this.webView = webView;
+    }
```

Comment:
I'm not really happy that the webview depends on the downloadlistener and the other way around, but this is also how the `FlutterWebChromeClient` works. This is less visible as that just uses a property of the parent class.